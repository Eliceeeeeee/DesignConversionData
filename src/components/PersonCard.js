import React from 'react';

const PersonCard = (props) => {
	return (
		<div className="personCard">
            <p>ID: {props.id}</p>
            <img src={props.imageUrl} id={props.id} alt={props.firstName}/>
            <p>{props.firstName} ,</p>
            <p>{props.lastName}</p>
        </div>
	);
}

export default PersonCard;