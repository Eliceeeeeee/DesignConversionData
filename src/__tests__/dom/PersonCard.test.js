import React from 'react';
import { shallow } from 'enzyme';
import PersonCard from '../../components/PersonCard';

describe('PersonCard should match the snapshot with conditions:', () => {
    it('has no props', () => {
        const component = shallow(<PersonCard />);
        expect(component).toMatchSnapshot();
    });

    it('has id props', () => {
        const component = shallow(<PersonCard id= "1" />);
        expect(component).toMatchSnapshot();
    });

    it('has imageUrl props', () => {
        const component = shallow(<PersonCard imageUrl="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg" />);
        expect(component).toMatchSnapshot();
    });

    it('has firstName props', () => {
        const component = shallow(<PersonCard firstName="George" />);
        expect(component).toMatchSnapshot();
    });

    it('has lastName props', () => {
        const component = shallow(<PersonCard lastName="Bluth" />);
        expect(component).toMatchSnapshot();
    });
});
