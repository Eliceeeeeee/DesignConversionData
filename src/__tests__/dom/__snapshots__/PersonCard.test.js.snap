// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard should match the snapshot with conditions: has firstName props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard
    firstName="George"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          ID: 
</p>,
        <img
          alt="George"
          id={undefined}
          src={undefined}
/>,
        <p>
          George
           ,
</p>,
        <p />,
      ],
      "className": "personCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ID: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "ID: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "George",
          "id": undefined,
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "George",
            " ,",
          ],
        },
        "ref": null,
        "rendered": Array [
          "George",
          " ,",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            ID: 
</p>,
          <img
            alt="George"
            id={undefined}
            src={undefined}
/>,
          <p>
            George
             ,
</p>,
          <p />,
        ],
        "className": "personCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ID: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ID: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "George",
            "id": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "George",
              " ,",
            ],
          },
          "ref": null,
          "rendered": Array [
            "George",
            " ,",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PersonCard should match the snapshot with conditions: has id props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard
    id="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          ID: 
          1
</p>,
        <img
          alt={undefined}
          id="1"
          src={undefined}
/>,
        <p>
           ,
</p>,
        <p />,
      ],
      "className": "personCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ID: ",
            "1",
          ],
        },
        "ref": null,
        "rendered": Array [
          "ID: ",
          "1",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "id": "1",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ,",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ,",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            ID: 
            1
</p>,
          <img
            alt={undefined}
            id="1"
            src={undefined}
/>,
          <p>
             ,
</p>,
          <p />,
        ],
        "className": "personCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ID: ",
              "1",
            ],
          },
          "ref": null,
          "rendered": Array [
            "ID: ",
            "1",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "id": "1",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ,",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ,",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PersonCard should match the snapshot with conditions: has imageUrl props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard
    imageUrl="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          ID: 
</p>,
        <img
          alt={undefined}
          id={undefined}
          src="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
/>,
        <p>
           ,
</p>,
        <p />,
      ],
      "className": "personCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ID: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "ID: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "id": undefined,
          "src": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ,",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ,",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            ID: 
</p>,
          <img
            alt={undefined}
            id={undefined}
            src="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
/>,
          <p>
             ,
</p>,
          <p />,
        ],
        "className": "personCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ID: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ID: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "id": undefined,
            "src": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ,",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ,",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PersonCard should match the snapshot with conditions: has lastName props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard
    lastName="Bluth"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          ID: 
</p>,
        <img
          alt={undefined}
          id={undefined}
          src={undefined}
/>,
        <p>
           ,
</p>,
        <p>
          Bluth
</p>,
      ],
      "className": "personCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ID: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "ID: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "id": undefined,
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ,",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ,",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bluth",
        },
        "ref": null,
        "rendered": "Bluth",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            ID: 
</p>,
          <img
            alt={undefined}
            id={undefined}
            src={undefined}
/>,
          <p>
             ,
</p>,
          <p>
            Bluth
</p>,
        ],
        "className": "personCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ID: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ID: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "id": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ,",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ,",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bluth",
          },
          "ref": null,
          "rendered": "Bluth",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PersonCard should match the snapshot with conditions: has no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          ID: 
</p>,
        <img
          alt={undefined}
          id={undefined}
          src={undefined}
/>,
        <p>
           ,
</p>,
        <p />,
      ],
      "className": "personCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ID: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "ID: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "id": undefined,
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ,",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ,",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            ID: 
</p>,
          <img
            alt={undefined}
            id={undefined}
            src={undefined}
/>,
          <p>
             ,
</p>,
          <p />,
        ],
        "className": "personCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ID: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ID: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "id": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ,",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ,",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
